### PR TITLE
feat: migrate trading endpoints

### PR DESCRIPTION
**Pull request by Rickk137**
https://sovryn.monday.com/boards/2218344956/pulses/2733453114

### CODE REVIEW

**Comment by soulBit** [comment]
Identified an issue with mainnet TVL data compared to live dapp, raised this with Shadow and will need to have that resolved before QA can test

File: src/app/pages/BuySovPage/components/InfoBar/index.tsx [hunk 1]
```diff
@@ -49,10 +39,12 @@ export function InfoBar() {
     value: { circulating_supply: totalSupply },
     loading: totalSupplyLoading,
   } = useFetch<{ circulating_supply: number; insertion_time: Date }>(
-    backendUrl[currentChainId] + '/sov/circulating-supply',
+    graphWrapperUrl[currentChainId] + '/sov/circulating-supply',
     { circulating_supply: 0 },
   );
 
+  console.log('circulating_supply:', totalSupply);
```
Comment (soulBit) [inline src/app/pages/BuySovPage/components/InfoBar/index.tsx:46]:
```suggestion
```

File: src/app/pages/LandingPage/index.tsx [hunk 2]
```diff
@@ -113,10 +111,11 @@ export const LandingPage: React.FC<ILandingPageProps> = ({
       cancelDataRequest.current = c;
     });
     axios
-      .get(url + '/api/v1/trading-pairs/summary', {
+      .get(graphWrapperUrl[currentChainId] + 'cmc/summary?extra=true', {
         cancelToken,
       })
       .then(res => {
+        console.log('pair:", res.data', res.data);
```
Comment (soulBit) [inline src/app/pages/LandingPage/index.tsx:118]:
```suggestion
```

**Review by soulBit — CHANGES_REQUESTED** [review]
Two small changes to make - I also spotted an inconsistency with the TVL data vs live dapp, I will query that on Monday story

File: src/app/hooks/useGetTokenPrice.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { bignumber } from 'mathjs';
+import { useMemo } from 'react';
+import { Asset } from 'types';
+import { APOLLO_POLL_INTERVAL } from 'utils/classifiers';
+import { AssetsDictionary } from 'utils/dictionaries/assets-dictionary';
+import { useGetTokenQuery } from 'utils/graphql/rsk/generated';
+
+export const useGetTokenPrice = (asset: Asset, type: 'BTC' | 'USD' = 'BTC') => {
```
Comment (tiltom) [inline src/app/hooks/useGetTokenPrice.tsx:8]: I don't like `type` typing - pls extract it to an enum that can stay in this file (you can call it `DenominationAsset`).

File: src/app/hooks/useGetTokenPrice.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  const price = useMemo(() => {
+    if (loading || !data) return '0';
+
+    const lastPrice =
+      type === 'BTC' ? data?.token?.lastPriceBtc : data?.token?.lastPriceUsd;
+
+    return bignumber(lastPrice || 0)
+      .mul(1e8)
+      .toFixed(4);
```
Comment (tiltom) [inline src/app/hooks/useGetTokenPrice.tsx:25]: I am not sure about this, I think you want to have the price as precise as possible and you only need to round it when displaying it.

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — CHANGES_REQUESTED** [review]
The code itself is fine but I see completely different data on testnet. Take a look at it pls and consult with @BetsyBraddock if needed. Also, as Betsy already mentioned in the story's comment, order crypto prices by marketcap.

New implementation:
image
image

Existing implementation:
image
image

**Review by tiltom — DISMISSED** [review]

**Review by Alitasovryn — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**